internal/bminventory/inventory.go

```diff
@@ -239,6 +241,37 @@ func (b *bareMetalInventory) updatePullSecret(pullSecret string, log logrus.Fiel
 	return pullSecret, nil
 }
 
+func mergeIgnitionConfig(base []byte, overrides []byte) (string, error) {
+	baseConfig, report, err := ign_3_1.Parse(base)
+	if err != nil {
+		return "", err
+	}
+	if report.IsFatal() {
+		return "", fmt.Errorf("base ignition config is invalid: %s", report.String())
```

Comment:
please use errors package and not fmt. errors contain stuck trace that can be used later.